ftp-utils/src/main/java/org/wso2/ei/b7a/ftp/core/client/FTPClient.java

```diff
@@ -137,22 +137,31 @@ public static void put(ObjectValue clientConnector, MapValue<Object, Object> inp
 
         try {
 
-            String url = FTPUtil.createUrl(clientConnector,
-                    inputContent.getStringValue(FTPConstants.INPUT_CONTENT_FILE_PATH_KEY));
             Map<String, String> propertyMap = new HashMap<>(
                     (Map<String, String>) clientConnector.getNativeData(FTPConstants.PROPERTY_MAP));
-            propertyMap.put(FTPConstants.PROPERTY_URI, url);
-
             boolean isFile = inputContent.getBooleanValue(FTPConstants.INPUT_CONTENT_IS_FILE_KEY);
+            boolean compressInput = inputContent.getBooleanValue(FTPConstants.INPUT_CONTENT_COMPRESS_INPUT_KEY);
+            InputStream stream = FTPClientHelper.getUploadStream(inputContent, isFile);
```

Comment:
Don't we have to close these resources at the end?

Reply:
Fixed.